src/es/hackstore/src/eu/kanade/tachiyomi/animeextension/es/hackstore/Hackstore.kt

```diff
@@ -208,47 +173,33 @@ class Hackstore : ConfigurableAnimeSource, ParsedAnimeHttpSource() {
 
     override fun videoListParse(response: Response): List<Video> {
         val document = response.asJsoup()
-        val videoList = mutableListOf<Video>()
-
-        val tabs = document.select("ul.TbVideoNv li.pres")
-
-        tabs.forEach { tab ->
+        return document.select("ul.TbVideoNv li.pres").parallelCatchingFlatMapBlocking { tab ->
             val server = tab.select("a.playr").text()
             val deco = tab.select("a.playr").attr("data-href")
             val langs = tab.select("a.playr").attr("data-lang")
             val fullUrl = baseUrl + deco
             val url = extractUrlFromDonFunction(fullUrl)
             val isLatino = langs.contains("latino")
             val isSub = langs.contains("subtitulado") || langs.contains("sub") || langs.contains("japonés")
+            val prefix = if (isLatino) "[LAT]" else if (isSub) "[SUB]" else "[CAST]"
 
             when {
-                server.contains("streamtape") -> {
-                    val video = StreamTapeExtractor(client).videoFromUrl(url, if (isLatino) "StreamTape Latino" else if (isSub) "StreamTape Subtitulado" else "StreamTape Castellano")
-                    if (video != null) {
-                        videoList.add(video)
-                    }
-                }
-                server.contains("voe") -> {
-                    val video = VoeExtractor(client).videosFromUrl(url, if (isLatino) "(Latino) " else if (isSub) "(Subtitulado) " else "(Castellano) ")
-                    videoList.addAll(video)
-                }
-                server.contains("filemoon") -> {
-                    val video = FilemoonExtractor(client).videosFromUrl(url, if (isLatino) "Filemoon Latino" else if (isSub) "Filemoon Subtitulado" else "Filemoon Castellano")
-                    videoList.addAll(video)
+                server.contains("streamtape") || server.contains("stp") || server.contains("stape") -> {
+                    listOf(StreamTapeExtractor(client).videoFromUrl(url, quality = "$prefix StreamTape")!!)
```

Comment:
Move out the extractors to a separate class instance here, and for hentaila as well